Formula/ghc.rb

```diff
@@ -48,6 +50,10 @@ class Ghc < Formula
   end
 
   def install
+    # Fix doc build error. Remove at version bump.
+    # https://gitlab.haskell.org/ghc/ghc/-/issues/19962
+    inreplace "docs/users_guide/conf.py", "'preamble': '''", "'preamble': r'''"
```

Comment:
Shouldn't we just use a patch based on 
https://gitlab.haskell.org/ghc/ghc/-/commit/83407ffc7acc00cc025b9f6ed063add9ab9f9bcc#2c8c4ca2eac2ad8b108ae37fa1b91a9f7676ca1e

e.g.
```diff
diff --git a/docs/users_guide/conf.py b/docs/users_guide/conf.py
index 83e0215..bdcf328 100644
--- a/docs/users_guide/conf.py
+++ b/docs/users_guide/conf.py
@@ -100,13 +100,13 @@ htmlhelp_basename = 'GHCUsersGuide'
 latex_elements = {
     'inputenc': '',
     'utf8extra': '',
-    'preamble': '''
+    'preamble': r'''
 \usepackage{fontspec}
 \usepackage{makeidx}
 \setsansfont{DejaVu Sans}
 \setromanfont{DejaVu Serif}
 \setmonofont{DejaVu Sans Mono}
-\setlength{\\tymin}{45pt}
+\setlength{\tymin}{45pt}
 ''',
 }
```

Reply:
We could. It's not clear how much of that commit we want to use, though (or if it even applied to the release tarball).

Reply:
Probably not the direct commit. For example, the diff I posted is based on release tar.

Reply:
Hmm, ok, yes, I think we want that extra change too. Will add it. Thanks for having a look.